src/tracker.rs

```diff
@@ -184,16 +223,25 @@ impl Tracker {
         }
     }
 
+    pub fn has_pointer(&self, point_offset: PointOffset) -> bool {
+        self.pending_updates.contains_key(&point_offset) || self.get(point_offset).is_some()
+    }
+
+    pub fn update(&mut self, point_offset: PointOffset, value_pointer: ValuePointer) {
+        self.pending_updates.insert(point_offset, value_pointer);
+    }
+
     /// Set value at the given point offset
     /// If the point offset is larger than the current length, the mapping is resized
-    pub fn set(&mut self, point_offset: PointOffset, value_pointer: ValuePointer) {
+    fn set(&mut self, point_offset: PointOffset, value_pointer: ValuePointer) {
         // save mapping to mmap
         self.persist_pointer(point_offset, Some(value_pointer));
         // increment header count if necessary
         self.increment_max_point_offset(point_offset);
     }
 
     pub fn unset(&mut self, point_offset: PointOffset) {
+        self.pending_updates.remove(&point_offset);
```

Comment:
Deletion from the tracker should also be persisted only on flush, so we need to have a separate storage for delete diff. See https://github.com/qdrant/qdrant/blob/9c20e9e27960228019b4606f137ca82b42fc3e66/lib/segment/src/common/rocksdb_buffered_update_wrapper.rs#L19

Reply:
Good observation. 

In the end I solved it using an enum within `pending_updates` so that we don't have to keep two collections for pending stuff in sync.